### ISSUE TITLE
"Timeout in connecting after 5000ms" error while connecting to mongo db.

### ISSUE THREAD

**Issue by schitransh** [strongloop/loopback-datasource-juggler#865]
HI 

I am facing a issue while fetching a records from mongo db.

Scenario.
we have a primary and geo cluster of mongo db. (4 application node instance and 4 mongo node instance.) [2 node in primary site and 2 node in geo site]
priority of mongo db primary to geo.

if we are terminating connectivity of primary site then geo application is not able to connect to db and getting below error but if i am login into mogodb it became primary in geo site. db execution is working properly.
how loopback-datasource-juggler will retry for connect?
is there any parameter need to set to connect dynamically?
```
2016-03-03 07:45:97 pid:14292 worker:2 {"msg":" Error occured while creation on composition records : Error: Timeout in connecting after 5000 ms","time":"2016-03-03T13:45:26.974Z","v":0}
```

Thanks
Chitransh

**Comment by ebarault** [comment]
did you check [this](https://docs.strongloop.com/display/public/LB/MongoDB+connector#MongoDBconnector-Replicasetconfiguration) ?

**Comment by loay** [comment]
Hi @schitransh 
Did you try @ebarault suggestion with the replica set configuration?

**Comment by loay** [comment]
Closing the issue for now. If it is still persisting, please feel free to leave a comment and I will re-open. Thanks.

**Comment by sanjaykumarns** [comment]
I am also facing the same issue.

> "loopback-connector-mongodb": "1.18.1"
> "loopback-datasource-juggler": "3.9.3"

```
Unhandled rejection Error: Timeout in connecting after 5000 ms
at Timeout._onTimeout (/node_modules/loopback-datasource-juggler/lib/datasource.js:2319:10)
at ontimeout (timers.js:482:11)
at tryOnTimeout (timers.js:317:5)
at Timer.listOnTimeout (timers.js:277:5)
```

**Comment by Milos5611** [comment]
I am also facing same one.
```
"loopback-connector-postgresql": "^3.9.0",
"loopback-datasource-juggler": "^4.19.2",

Cannot start the application. Error: Timeout in connecting after 5000 ms
Timeout._onTimeout (/home/node/api/node_modules/loopback-datasource-juggler/lib/datasource.js:2640:10)
at ontimeout (timers.js:436:11)
at tryOnTimeout (timers.js:300:5)
at listOnTimeout (timers.js:263:5)
 at Timer.processTimers (timers.js:223:10)


```

**Comment by emonddr** [comment]
@Milos5611 , you are using

```
loopback-connector-postgresql"
```

not mongdodb.

Please provide a minimal LB3 or LB4 application that reproduces your problem, to help us investigate.
Thanks.

**Comment by emonddr** [comment]
> I am also facing the same issue.
> 
> > "loopback-connector-mongodb": "1.18.1"
> > "loopback-datasource-juggler": "3.9.3"
> 
> ```
> Unhandled rejection Error: Timeout in connecting after 5000 ms
> at Timeout._onTimeout (/node_modules/loopback-datasource-juggler/lib/datasource.js:2319:10)
> at ontimeout (timers.js:482:11)
> at tryOnTimeout (timers.js:317:5)
> at Timer.listOnTimeout (timers.js:277:5)
> ```

@sanjaykumarns ,

Is your setup as complex as @schitransh ? Or are you connecting to one mongodb instance?
Please provide a minimal LB3 or LB4 application that reproduces your issue to help us investigate.
Thank you.

**Comment by danysz** [comment]
the problem still persist

https://github.com/danysz/test-connection-app-lb4-pg

**Comment by PabloLionel** [comment]
The issue should have been fixed by #1585.
